pkg/apis/kubeone/validation/validation_test.go

```diff
@@ -1028,25 +1042,90 @@ func TestValidateClusterNetworkConfig(t *testing.T) {
 			clusterNetworkConfig: kubeoneapi.ClusterNetworkConfig{
 				PodSubnet:     "192.168.1.0/24",
 				ServiceSubnet: "192.168.0.0",
+				IPFamily:      kubeoneapi.IPFamilyIPv4,
 			},
 			expectedError: true,
 		},
 		{
 			name: "invalid cni config",
 			clusterNetworkConfig: kubeoneapi.ClusterNetworkConfig{
+				IPFamily: kubeoneapi.IPFamilyIPv4,
 				CNI: &kubeoneapi.CNI{
 					Canal:    &kubeoneapi.CanalSpec{},
 					WeaveNet: &kubeoneapi.WeaveNetSpec{},
 				},
 			},
 			expectedError: true,
 		},
+		{
+			name: "valid ipv6 config",
+			clusterNetworkConfig: kubeoneapi.ClusterNetworkConfig{
+				IPFamily:             kubeoneapi.IPFamilyIPv6,
+				PodSubnet:            "fd01::/48",
+				ServiceSubnet:        "fd02::/120",
+				NodeCIDRMaskSizeIPv6: ptr(64),
+			},
+		},
+		{
+			name: "valid ipv4+ipv6 config",
+			clusterNetworkConfig: kubeoneapi.ClusterNetworkConfig{
+				IPFamily:             kubeoneapi.IPFamilyIPv4IPv6,
+				PodSubnet:            "10.244.0.0/16,fd01::/48",
+				ServiceSubnet:        "10.96.0.0/12,fd02::/120",
+				NodeCIDRMaskSizeIPv4: ptr(24),
+				NodeCIDRMaskSizeIPv6: ptr(64),
+			},
+		},
+		{
+			name: "valid ipv6+ipv4 config",
+			clusterNetworkConfig: kubeoneapi.ClusterNetworkConfig{
+				IPFamily:             kubeoneapi.IPFamilyIPv6IPv4,
+				PodSubnet:            "fd01::/48,10.244.0.0/16",
+				ServiceSubnet:        "fd02::/120,10.96.0.0/12",
+				NodeCIDRMaskSizeIPv4: ptr(24),
+				NodeCIDRMaskSizeIPv6: ptr(64),
+			},
+		},
+		{
+			name: "invalid ipv6+ipv4 config",
+			clusterNetworkConfig: kubeoneapi.ClusterNetworkConfig{
+				IPFamily:             kubeoneapi.IPFamilyIPv6IPv4,
+				PodSubnet:            "10.244.0.0/16,fd01::/48",
+				ServiceSubnet:        "10.96.0.0/12,fd02::/120",
+				NodeCIDRMaskSizeIPv4: ptr(24),
+				NodeCIDRMaskSizeIPv6: ptr(64),
+			},
+			expectedError: true,
+		},
+		{
+			name: "invalid nod cidr mask size",
+			clusterNetworkConfig: kubeoneapi.ClusterNetworkConfig{
+				IPFamily:             kubeoneapi.IPFamilyIPv4IPv6,
+				PodSubnet:            "10.244.0.0/16,fd01::/48",
+				ServiceSubnet:        "10.96.0.0/12,fd02::/120",
+				NodeCIDRMaskSizeIPv4: ptr(16),
+				NodeCIDRMaskSizeIPv6: ptr(48),
+			},
+			expectedError: true,
+		},
+		{
+			name: "invalid number of cidrs",
+			clusterNetworkConfig: kubeoneapi.ClusterNetworkConfig{
+				IPFamily:             kubeoneapi.IPFamilyIPv4IPv6,
+				PodSubnet:            "10.244.0.0/16",
+				ServiceSubnet:        "10.96.0.0/12,fd02::/120",
+				NodeCIDRMaskSizeIPv4: ptr(16),
+				NodeCIDRMaskSizeIPv6: ptr(48),
+			},
+			expectedError: true,
+		},
 	}
 	for _, tc := range tests {
 		tc := tc
 		t.Run(tc.name, func(t *testing.T) {
 			errs := ValidateClusterNetworkConfig(tc.clusterNetworkConfig, nil)
 			if (len(errs) == 0) == tc.expectedError {
+				t.Log(errs)
```

Comment:
Is this a leftover?

Reply:
No. I added it intentionally to know what exactly went wrong in case of error.